plugins/chat/app/jobs/regular/chat/update_chat_threads_enabled.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Jobs
+  module Chat
+    class UpdateChatThreadsEnabled < Jobs::Base
+      def execute(args = {})
+        has_thread_channels = ::Chat::Channel.where(threading_enabled: true).exists?
+        return if SiteSetting.chat_threads_enabled === has_thread_channels
+
+        SiteSetting.chat_threads_enabled = has_thread_channels
```

Comment:
Can't we just do this and bypass the check above? I would think setting a site setting is a no-op if we don't change the value.